### ISSUE TITLE
About the loss

### ISSUE THREAD

**Issue by jeniurs** [TranQuocTrinh/transformer#3]
Hello. I'm the one that just sent a question about your code via twitter.
There seems to be a mismatch of size, and content between the ground truth and the outputs of the model.
The ground truth(target) is the size of 5100 and it looks like integer tensors (integers set by the tokenizer rules), but the output of the model is 64000 size of possibilities. How can the loss be computed? 
Thanks.

**Comment by tranquoctrinh** [comment]
@jeniurs 
```bash
     55         loss = F.cross_entropy(preds.view(-1, preds.size(-1)), gold, ignore_index=target_pad_id)
     56         loss.backward()

ipdb> preds.view(-1, preds.size(-1)).shape
torch.Size([5100, 64000])
ipdb> gold.shape
torch.Size([5100])
ipdb> loss = F.cross_entropy(preds.view(-1, preds.size(-1)), gold, ignore_index=target_pad_id)
ipdb> loss
tensor(11.0623, device='cuda:0', grad_fn=<NllLossBackward0>)
```

Exactly, gold is labeled and it has size 5100, and preds is a tensor with size (5100, 64000). You can refer to how to compute the loss in this https://pytorch.org/docs/stable/generated/torch.nn.functional.cross_entropy.html
